api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
I think if @pmorie and I can't easily explain the pattern we were proposing in a way that makes you go "ah ha!" then maybe it's not the best path. Do you think it makes more intuitive sense to just blend REST and RPC? So like, get the current debugger state with `GET /state`:

```go
type DebuggerState struct {
    // Running is true if the debugged process is continuing execution, and is
    // false if the debugged process is currently suspended.
    Running bool `json:"running"`
    // BreakPoint is the current breakpoint at which the debugged process is
    // suspended, and may be empty if the process is not suspended.
    BreakPoint BreakPoint `json:"breakPoint,omitempty"`
    // Thread is the currently selected debugger thread.
    Thread Thread `json:"thread"`
}
```

And an RPC-like endpoint for commands, sort of like an earlier API revision I had proposed. `POST /command`:

```go
type DebuggerCommand struct {
    Name string // like continue, step, next
}
```

Which could return just a generic status code/message, or maybe the new `DebuggerState`.

Thoughts?